### PR TITLE
Documentation restructure

### PR DESCRIPTION
**Pull request by DougManuel**
Restructured vignettes following addition/fixes for flexible NA handling. Aim for more expandability as we add features and a quick start for minimal use.

### Documentation restructure
  - **Getting started vignette**: Introductory
  tutorial with simple examples for new users
  - **User guide**: Comprehensive feature
  documentation with all MockData capabilities
  - **Advanced topics**: Technical details on NA
  handling internals, derived variables, and
  custom metadata
  - **How-to guides**: Expanded CCHS, CHMS, and
  DemPoRT examples with realistic workflows
  - **pkgdown configuration**: Added
  `_pkgdown.yml` for documentation website
  generation
  - **README updates**: Refreshed to reference
  new vignette structure

  ### Technical improvements
  - Enhanced NA data handling for special codes
  (996-999)
  - Improved test coverage for variable category
  extraction
  - Fixed `get_variable_categories()` to handle
  'else' conditions in recStart

  ### Documentation organization
Structured learning path
  1. Getting started → basics
  2. User guide → complete feature reference
  3. How-to guides → domain-specific examples
  (CCHS/CHMS/DemPoRT)
  4. Advanced topics → internals and edge cases

  ## Testing

  - All vignettes render successfully
  - Added tests for NA handling edge cases
  - Verified examples work with real metadata

  ## Files changed

  - New: `getting-started.qmd`, `user-guide.qmd`,
   `advanced-topics.qmd`, `_pkgdown.yml`
  - Enhanced: CCHS, CHMS, DemPoRT example
  vignettes
  - Updated: README.md

  ## Next steps

  After merge, consider:
  - Publishing pkgdown site to GitHub Pages
  - Adding vignette cross-references
  - Creating quick reference card

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Package: MockData
 Title: Generate Mock Data from Recodeflow Metadata
-Version: 0.2.0
+Version: 0.2.1
 Authors@R: c(
     person("Juan", "Li", role = c("aut", "cre"), email = "juan.li@oahpp.ca"),
```
Comment (JuanLiOHRI) [inline DESCRIPTION:5]: My email is juli@ohri.ca

**Review by JuanLiOHRI — APPROVED** [review]
Looks good to me.

File: vignettes/chms-example.qmd [hunk 2]
```diff
@@ -1,199 +1,390 @@
...
-cat("\nFirst 10 rows:\n")
+# Display first rows
+cat("First 10 rows:\n")
 print(head(df_mock, 10))
 
-# Check for missing values
-cat("\nMissing value summary:\n")
+# Check missing data
+cat("\nMissing value counts:\n")
 na_counts <- colSums(is.na(df_mock))
```
Comment (rafdoodle) [inline vignettes/chms-example.qmd:189]: This counts the number of actual NAs in the data frame, not the number of missing data codes which are actually in the generated mock data. Might need to be addressed across all vignettes.

File: vignettes/demport-example.qmd [hunk 3]
```diff
@@ -1,197 +1,413 @@
...
+  for (check_name in names(checks)) {
+    status <- ifelse(checks[[check_name]], "\u2713", "\u2717")
+    cat(status, check_name, "\n")
+  }
+
+  return(all_pass)
+}
+
+# Validate our mock data
+validate_demport_data(df_mock)
```
Comment (rafdoodle) [inline vignettes/demport-example.qmd:391]: Returns `✗ has_age `

**Review by rafdoodle — COMMENTED** [review]
The mock data generation looks good to me! I corrected the chms metadata and qmd, and have a few small comments regarding the vignettes.

**Comment by DougManuel** [comment]
I've incorporated changes in this PR into PR #8. I'll close this PR.
